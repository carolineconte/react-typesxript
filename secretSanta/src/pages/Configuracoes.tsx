import styled from "styled-components"
import ListaParticipantes from "../components/ListaParticipantes"
import Rodape from "../components/Rodape"
import Formulario from "../components/Formulario"

const ContainerStyled = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  gap: 2rem;
  width: 100%;
`

const Configuracoes = () => {
  return (
    <ContainerStyled>
      <Formulario />
      <ListaParticipantes />
      <Rodape />
    </ContainerStyled>
  )
}

export default Configuracoes