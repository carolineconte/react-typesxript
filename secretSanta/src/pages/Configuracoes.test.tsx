import { render } from "@testing-library/react"
import { RecoilRoot } from "recoil"
import Configuracoes from "./Configuracoes"

const mockNavegacao = jest.fn()
jest.mock('react-router-dom', () => {
  return{
    // Para mockar qualque hook, a implementacao deve ser assim
    useNavigate: () => mockNavegacao
  }
  
})

describe('a pagina padrao', () => {
  test('deve ser renderizada corretamente', () => {
    const { container } = render(
      <RecoilRoot>
        <Configuracoes/>
      </RecoilRoot>
    )
    expect(container).toMatchSnapshot()
  })
})