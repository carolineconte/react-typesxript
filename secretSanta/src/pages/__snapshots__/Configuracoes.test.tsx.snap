// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`a pagina padrao deve ser renderizada corretamente 1`] = `
<div>
  <div
    class="sc-gLLuof ljktgW"
  >
    <form
      class="sc-dmyCSP Jcllc"
    >
      <input
        class="sc-hLQSwg hiYdJT"
        placeholder="Insira os nomes dos participantes"
        type="text"
        value=""
      />
      <button
        class="sc-eDLKkx YOyNh"
        disabled=""
      >
        Adicionar
      </button>
      
    </form>
    <ul />
    <footer
      class="sc-beySPh dgeLs"
    >
      <button
        class="sc-guDLey ctVcAp"
        disabled=""
      >
        Iniciar brincadeira
      </button>
      <img
        alt="Sacolas de compras"
        src="sacolas.png"
      />
    </footer>
  </div>
</div>
`;
